modules/app.rb

```diff
@@ -0,0 +1,166 @@
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'classroom'
+require_relative 'person'
+
+class App
+  attr_accessor :books, :rentals, :people
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def selected_option(option)
+    case option
+
+    when '1'
+      list_books
+    when '2'
+      list_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals
+    else
+      'Please select valid option from the list!'
+    end
+  end
+
+  def list_books
+    @books.each do |book|
+      book.instance_variables.each do |var|
+        text = ''
+        value = book.instance_variable_get(var)
+        var = var.to_s.delete('@')
+        var = var.capitalize
+        text += "#{var}:#{value}" unless var.include?('Rentals')
+        puts text
+      end
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      if person.is_a?(Student)
+        puts "[Student] Name: #{person.name}, ID: #{person.object_id}, Age: #{person.age}"
+      elsif person.is_a?(Teacher)
+        puts "[Teacher] Name: #{person.name}, ID: #{person.object_id}, Age: #{person.age}"
+      else
+        puts "Name: #{person.name}"
+      end
+      puts "\n"
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student(1) or a teacher(2)?'
+    choise = gets.chomp
+
+    if choise == '1'
+      create_student
+    elsif choise == '2'
+      create_teacher
+    else
+      puts 'Your input is wrong!'
+    end
+  end
+
+  def create_student
+    puts 'Age:'
+    age = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Has parent permission? [Y/N]'
+    parent_permission = gets.chomp.upcase == 'Y'
+    student = Student.new(age, name, parent_permission)
+    @people.push(student)
+    puts 'You added student successfully'
+  end
+
+  def create_teacher
+    puts 'Age:'
+    age = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Specialization:'
+    specialization = gets.chomp
+    teacher = Teacher.new(age, name, specialization)
+    @people.push(teacher)
+    puts 'You added teacher successfully'
+  end
```

Comment:
Nice job so far👍. I think it is currently impossible to create a teacher🥲 When I run the command to create a teacher, an ArgumentError is logged into the console, terminating the process😉. Kindly fix this bug please as this can lead to a bad user experience😉.

<img width="525" alt="image" src="https://github.com/ikoote1/school-library/assets/64643995/1a7af362-bdfe-4748-946f-77ef18cca163">